### PR TITLE
add libobjcryst

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
It would be good to have some tests for this.

**Comment by ericdill** [comment]
@cj-wright what do you mean "tests"

**Comment by CJ-Wright** [comment]
Right now there is only a dummy test for this, so something (even checking that compiled files were written).

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes can I get some review/merge?

File: recipes/libobjcryst/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{% set md5 = "08d33a1fb654a6fec9c16ddccbbc8c72" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/diffpy/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: Use `sha256`

File: recipes/libobjcryst/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - scons
+    - boost 1.61.0
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: Add `toolchain` here

File: recipes/libobjcryst/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - true
+
+about:
+  home: https://github.com/diffpy/libobjcryst/
+  license: GPL2
+  license_family: GPL
+  license_file: LICENSE.txt
+  summary: 'ObjCryst++ packaged for installation as a shared library.'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: Remove this

File: recipes/libobjcryst/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  skip: True  # [win]
+
+requirements:
+  build:
+    - scons
+    - boost 1.61.0
+
+test:
+  commands:
+    - true
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: Add a test like `test -f $PREFIX/lib/libObjCryst${SHLIB_EXT}` here

File: recipes/libobjcryst/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{% set name = "libobjcryst" %}
+{% set version = "2017.1.1" %}
+{% set sha256 = "513db8853fe306d6620969fd8e82cff179c62baf5d44957b0eca8983528d4f5a" %}
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: This value appears to be wrong

**Comment by CJ-Wright** [comment]
Pending https://github.com/diffpy/libobjcryst/issues/5

**Comment by isuruf** [comment]
If you need python 2, add `python # [py2k]` as a build requirement, but don't add it for runtime.

**Comment by isuruf** [comment]
If you need python 2, add `python` as a build requirement and skip `py3k`, but don't add it for runtime.

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes ready for more review/merge!

File: recipes/libobjcryst/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    library. libobjcryst does not include GUI related files from ObjCryst++.
+    This distribution contains required sources from cctbx and newmat that are
+    used in ObjCryst++.
+  doc_url: https://github.com/diffpy/libobjcryst
+  dev_url: https://github.com/diffpy/libobjcryst
+
+extra:
+  recipe-maintainers:
+    - CJ-Wright
+    - pavoljuhas
```
Comment (isuruf) [inline recipes/libobjcryst/meta.yaml]: @pavoljuhas, are you ok with being added as a maintainer for this recipe

**Comment by CJ-Wright** [comment]
Thank you!

File: recipes/libobjcryst/meta.yaml (hunk 6 above)
Comment (pavoljuhas) [reply recipes/libobjcryst/meta.yaml]: @isuruf - Yes, that is OK with me.

File: recipes/libobjcryst/build.sh [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+export CPATH="$PREFIX/include:$CPATH"
+MYNCPU=$(( (CPU_COUNT > 8) ? 8 : CPU_COUNT ))
+
+scons -j $MYNCPU
+scons install prefix=$PREFIX
```
Comment (pavoljuhas) [inline recipes/libobjcryst/build.sh]: Can you add newline to the end of file?

**Pull request merged**